### PR TITLE
Align y-axes in mmm.plot.budget_allocation

### PR DESCRIPTION
**Pull request by TeemuSailynoja**
## Description

Ensure that y=0 is aligned between the primary and secondary y-axis in `mmm.plot.budget_allocation`.
## Solution
If one axis includes negative, values the other is extended down so that the y=0 is aligned.

If all contributions would be fully negative (I don't know when this would happen...), the y=0 level is centered vertically on both axes.

## Related Issue

- [ ] Closes #
- [ ] Related to #

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://www.pymc-marketing.io/en/latest/contributing/index.html). Feel free to comment [`pre-commit.ci autofix` to auto-fix](https://pre-commit.ci/#configuration-autofix_prs).
- [ ] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks) using [numpydoc format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [x] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

----
📚 Documentation preview 📚: https://pymc-marketing--1967.org.readthedocs.build/en/1967/

### CODE REVIEW

**Comment by TeemuSailynoja** [comment]
still needs tests to go through the combinations of negative positive and ensure that the axis limits are correct after the call.

File: pymc_marketing/mmm/plot.py [hunk 1]
```diff
@@ -243,6 +243,40 @@ def _build_subplot_title(
...
+            if ax1_yratio < ax2_yratio:
+                ax2.set_ylim(bottom=ylims2[1] * ax1_yratio)
+                if ax1_yratio == -1:
+                    # if the axis is fully negative, center zero.
+                    ax.set_ylim(top=-ylims1[0])
+            elif ax2_yratio < ax1_yratio:
+                ax.set_ylim(bottom=ylims1[1] * ax2_yratio)
+                if ax2_yratio == -1:
+                    # if the axis is fully negative, center zero.
+                    ax2.set_ylim(top=-ylims2[0])
```
Comment (williambdean) [inline pymc_marketing/mmm/plot.py]: Can you use NamedTuple or something. Quite confusing to read

Comment (TeemuSailynoja) [reply pymc_marketing/mmm/plot.py]: Looking at this now, I fully agree. I also think that this is not capturing every possible combination the axes can get odd. I'll move this to a draft and rework it.

Comment (TeemuSailynoja) [reply pymc_marketing/mmm/plot.py]: I completely rewrote the function, and added tests that go through the possible cases.
Not the most compact solution, but I hope it exposes its logic better.

File: pymc_marketing/mmm/plot.py [hunk 2]
```diff
@@ -243,6 +244,83 @@ def _build_subplot_title(
             return ", ".join(title_parts)
         return fallback_title
 
+    def _align_y_axes(self, ax_left, ax_right) -> None:
```
Comment (juanitorduz) [inline pymc_marketing/mmm/plot.py]: @TeemuSailynoja can you please add type hinds and complete docstrings to this function so that we can bring it to the finish line :) Thanks!

**Review by juanitorduz — CHANGES_REQUESTED** [review]

File: pymc_marketing/mmm/plot.py (hunk 2 above)
Comment (TeemuSailynoja) [reply pymc_marketing/mmm/plot.py]: @juanitorduz I'm sorry for the delay. Work got in the way 😝 

I hope what I added is good. Thank you for pingin me on this!

**Review by juanitorduz — APPROVED** [review]
